common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -355,18 +370,22 @@ public boolean doMouseClicked(double mouseX, double mouseY, int button) {
                 if (hovered.hasStaticLocation()) {
                     if (hovered instanceof IconPoi iconPoi) {
                         if (iconPoi instanceof CustomPoi customPoi) {
-                            Models.Compass.setCompassLocation(
-                                    new Location(hovered.getLocation()), iconPoi.getIcon(), customPoi.getColor());
+                            Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation(
+                                    new Location(hovered.getLocation()),
+                                    iconPoi.getIcon(),
+                                    customPoi.getColor(),
+                                    customPoi.getColor());
                         } else {
-                            Models.Compass.setCompassLocation(new Location(hovered.getLocation()), iconPoi.getIcon());
+                            Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation(
+                                    new Location(hovered.getLocation()), iconPoi.getIcon());
                         }
                     } else {
-                        Models.Compass.setCompassLocation(new Location(hovered.getLocation()));
+                        Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation(new Location(hovered.getLocation()));
                     }
                 } else {
                     final Poi finalHovered = hovered;
-                    Models.Compass.setDynamicCompassLocation(
-                            () -> finalHovered.getLocation().asLocation());
+                    Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation(new DynamicLocationSupplier(
```

Comment:
I'm losing my way among all those else clauses, but I guess the same applies to these.